### ISSUE TITLE
Script to convert annotation xml files to .txt file

### ISSUE THREAD

**Issue by kmsravindra** [AlexeyAB/darknet#864]
@AlexeyAB ,

I have used darkflow earlier and I wanted to make a switch to your library as this performs great with lot of options! So, I already have several annotations that were done in .xml...The content looks like the below - Is there a script than can simply convert it to .txt file without having to use Yolo_mark? Otherwise I can write a script for myself, if you can tell me what I need to take care of. Your help is much appreciated!

----------------------
<annotation>
<folder>C:/Users/kmsra_000/Documents/Machine_Learning/Experiments/Yolo_object_detection/darkflow-master/vid_images_20fps</folder>
  <filename>frame223.png</filename>
  <segmented>0</segmented>
  <size>
    <width>544</width>
    <height>464</height>
    <depth>3</depth>
  </size>
  <object>
    <name>suspect_polyp</name>
    <pose>Unspecified</pose>
    <truncated>0</truncated>
    <difficult>0</difficult>
    <bndbox>
      <xmin>56</xmin>
      <ymin>178</ymin>
      <xmax>190</xmax>
      <ymax>302</ymax>
    </bndbox>
  </object>
</annotation>

**Comment by AlexeyAB** [comment]
Your xml-file looks like default format such as in the Pascal VOC, so you should try to use: https://github.com/AlexeyAB/darknet/blob/master/scripts/voc_eval.py
(or for Python 3: https://github.com/AlexeyAB/darknet/blob/master/scripts/voc_eval_py3.py )

----

Yolo format: `class_id relative_x_center relative_y_center relative_width relative_height`
txt-file should be like this: 
```
1 0.716797 0.395833 0.216406 0.147222
0 0.687109 0.379167 0.255469 0.158333
1 0.420312 0.395833 0.140625 0.166667
```
Convertation:
`yolo_x = (xmin+xmax)/2.0`
`yolo_y = (ymin+ymax)/2.0`
`yolo_width = xmax - xmin`
`yolo_height = ymax - ymin`

**Comment by kmsravindra** [comment]
@AlexeyAB, If i have only one class, then the class_id is 0 ?

**Comment by kmsravindra** [comment]
Here is the python code that I created, if someone is interested - 
[https://github.com/kmsravindra/useful_scripts/blob/master/Convert_PascalVOC_XML_to_darknet_text.py](url)

Note - I have not found 'class' attribute in pascal voc xml file (pls let me know if I am missing something). So, for now I have hardcoded the class as 0 as I only have to deal with a single class. In case you have multiple classes, one work around is to modify the above script to create a new class for each unique object_name that is read from the xml file.  If you have any better suggestions, then let me know. Thanks!

**Comment by AlexeyAB** [comment]
@kmsravindra If the image size is changed in pixels, then you should change (x,y,w,h)-of objects.

Can you provide examples of image? before, after

**Comment by kmsravindra** [comment]
Please find the images below. 
1. The red box is the bounding box
2. green circle is the object
3. black border is the padding done to all input images (of different aspects and sizes) into a uniform size and aspect ratio.

So I need to accordingly transform the annotations done on such input images when we transform the images. 

Previous image - 
![previous_image](https://user-images.githubusercontent.com/3147499/40480665-d09022d0-5f6c-11e8-990d-176da5880e23.png)

Padded image with black border ( notice that the aspect ratio and the size of the image changed but the object is not distorted)
![padded_image](https://user-images.githubusercontent.com/3147499/40480709-f0f2bcfe-5f6c-11e8-9549-c8c9f72265f1.png)

**Comment by AlexeyAB** [comment]
Yes, you should recompute all in this case: x_yolo, y_yolo , yolo_width and yolo_height.

Because image_size is changed, then relative sizes will be changed too.

**Comment by kmsravindra** [comment]
Thank you very much

**Comment by sinaikh** [comment]
hi 

i downloaded a data set that just contain  .xml files (voc format) and jpeg images
i trying to convert it to yolo format , i use this script    "https://github.com/AlexeyAB/darknet/blob/master/scripts/voc_label.py"

but i get this error 

"C:\Users\Mascia\Desktop\deep_learning\VOCdevkit>python.exe voc_label.py
Traceback (most recent call last):
  File "voc_label.py", line 50, in <module>
    image_ids = open('VOCdevkit/VOC%s/ImageSets/Main/%s.txt'%(year, image_set)).read().strip().split()
IOError: [Errno 2] No such file or directory: 'VOCdevkit/VOC2012/ImageSets/Main/train.txt'"

What should i do?
thank you for your help

**Comment by sharoseali** [comment]
** @sinaikh **
It seems u have downloaded test set not for training .. Download from the link below and then try..
http://pjreddie.com/media/files/VOCtrainval_06-Nov-2007.tar

**Comment by MuhammadAsadJaved** [comment]
@kmsravindra  @dexception Have you wrote the script for this task? I also have almost similar problem. I have cropped my original images and now i want to update the original annotation file according to the cropped regions.  I convert 1 image to 4 new images by cropping them to equal size. So How can i convert one original annotation file to 4 new annotation files according to the cropped regions?  I have both XML and txt annotation files.
